test/extended/operators/server_side_apply_client_copy.go

```diff
@@ -0,0 +1,198 @@
+	}
+	desiredConfiguration := applyoperatorv1.StaticPodOperatorStatus()
+	desiredConfiguration.OperatorStatusApplyConfiguration = *applyConfiguration
+	return c.applyOperatorStatus(ctx, fieldManager, desiredConfiguration)
+}
+
+func (c dynamicOperatorClient) applyOperatorStatus(ctx context.Context, fieldManager string, desiredConfiguration *applyoperatorv1.StaticPodOperatorStatusApplyConfiguration) (string, error) {
+	applyMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(desiredConfiguration)
+	if err != nil {
+		return "", fmt.Errorf("failed to convert to unstructured: %w", err)
+	}
+	applyUnstructured := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"status": applyMap,
+		},
+	}
+	applyUnstructured.SetGroupVersionKind(c.gvk)
+	applyUnstructured.SetName(c.configName)
+
+	original, err := c.client.Get(ctx, c.configName, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		// do nothing and proceed with the apply
+	case err != nil:
+		return "", fmt.Errorf("unable to read existing %q: %w", c.configName, err)
+	default:
+		if c.extractApplyStatus == nil {
+			return "", fmt.Errorf("extractApplyStatus is nil")
+		}
+		previouslySetFields, err := c.extractApplyStatus(original, fieldManager)
+		if err != nil {
+			return "", fmt.Errorf("unable to extract status for %q: %w", fieldManager, err)
+		}
+		if equality.Semantic.DeepEqual(previouslySetFields, desiredConfiguration) {
+			// nothing to apply, so return early
+			return "nothing to apply", nil
+		}
+	}
+
+	_, err = c.client.ApplyStatus(ctx, c.configName, applyUnstructured, metav1.ApplyOptions{
+		Force:        true,
+		FieldManager: fieldManager,
+	})
+	if err != nil {
+		return "modification attempted", fmt.Errorf("unable to ApplyStatus for operator using fieldManager %q: %w", fieldManager, err)
+	}
+
+	return "modification done", nil
+}
+
+func extractOperatorSpec(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
+	castObj := &operatorv1.OpenShiftAPIServer{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to OpenShiftAPIServer: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractOpenShiftAPIServer(castObj, fieldManager)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract fields for %q: %w", fieldManager, err)
+	}
+	if ret.Spec == nil {
+		return nil, nil
+	}
+	return &ret.Spec.OperatorSpecApplyConfiguration, nil
+}
+
+func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
+	castObj := &operatorv1.OpenShiftAPIServer{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to OpenShiftAPIServer: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractOpenShiftAPIServerStatus(castObj, fieldManager)
```

Comment:
I messed up this name.

Reply:
ah, so instead of extracting the status you were extracting the spec thus the diff!